src/Polly.Core/Strategy/OutcomePredicate.TResult.cs

```diff
@@ -0,0 +1,222 @@
+    /// <param name="predicate">The predicate to determine if the exception should be retried.</param>
+    /// <returns>The current updated instance.</returns>
+    public OutcomePredicate<TArgs, TResult> HandleException<TException>(Func<TException, bool> predicate)
+        where TException : Exception
+    {
+        Guard.NotNull(predicate);
+
+        return HandleOutcome((outcome, _) =>
+        {
+            if (outcome.Exception is TException typedException)
+            {
+                return new ValueTask<bool>(predicate(typedException));
+            }
+
+            return new ValueTask<bool>(false);
+        });
+    }
+
+    /// <summary>
+    /// Adds an exception predicate for the specified exception type.
+    /// </summary>
+    /// <typeparam name="TException">The exception type to add a predicate for.</typeparam>
+    /// <param name="predicate">The predicate to determine if the exception should be retried.</param>
+    /// <returns>The current updated instance.</returns>
+    public OutcomePredicate<TArgs, TResult> HandleException<TException>(Func<TException, TArgs, bool> predicate)
+        where TException : Exception
+    {
+        Guard.NotNull(predicate);
+
+        return HandleOutcome((outcome, args) =>
+        {
+            if (outcome.Exception is TException typedException)
+            {
+                return new ValueTask<bool>(predicate(typedException, args));
+            }
+
+            return new ValueTask<bool>(false);
+        });
+    }
+
+    /// <summary>
+    /// Adds an exception predicate for the specified exception type.
+    /// </summary>
+    /// <typeparam name="TException">The exception type to add a predicate for.</typeparam>
+    /// <param name="predicate">The predicate to determine if the exception should be retried.</param>
+    /// <returns>The current updated instance.</returns>
+    public OutcomePredicate<TArgs, TResult> HandleException<TException>(Func<TException, TArgs, ValueTask<bool>> predicate)
+        where TException : Exception
+    {
+        Guard.NotNull(predicate);
+
+        return HandleOutcome((outcome, args) =>
+        {
+            if (outcome.Exception is TException typedException)
+            {
+                return predicate(typedException, args);
+            }
+
+            return new ValueTask<bool>(false);
+        });
+    }
+
+    /// <summary>
+    /// Adds a result predicate for the specified result value.
+    /// </summary>
+    /// <param name="value">The result value to be retried.</param>
+    /// <returns>The current updated instance.</returns>
+    /// <remarks>
+    /// By default, the default equality comparer is used to compare the result value with the value specified in this method.
+    /// </remarks>
+    public OutcomePredicate<TArgs, TResult> HandleResult(TResult value)
+    {
+        return HandleResult((result, _) => new ValueTask<bool>(EqualityComparer<TResult>.Default.Equals(result!, value)));
```

Comment:
Should we provide an overload that allows for a user-specified equality comparer, or are the `Func<TResult, bool>` overloads enough to allow a user to do that themselves with: this?

```csharp
var myComparer = ...;
HandleResult((result, _) => new ValueTask<bool>(myComparer.Equals(result!, value)
```

Adding the overload makes it simpler and avoids the (extra?) closure, but having the overload(s) increases the API surface area/complexity.

Reply:
I have added optional comparer parameter. I think it's fine and provides greater flexibility.